### PR TITLE
Make error properties non-enumerable

### LINKED ISSUE #29: Consider making common error properties non enumerable
## Description

Native JS errors don't have its properties enumerable. What this means is that the `name`, `stack` and `message` properties do not appear when console.logging errors.

Notice the `name` and `message` property appearing below.

image

## Proposal

To get deserialized errors closer to the native ones, when deserializing the error, we could use `Object.defineProperty` instead of a direct assignment to make the properties not enumerable. eg: here https://github.com/sindresorhus/serialize-error/blob/master/index.js#L54

```ts
for (const property of commonProperties) {
	if (typeof from[property] === 'string') {
        Object.defineProperty(to, property, {
          value: from[property]
        })
	}
}
```

I'm happy to make a PR if you agree with the proposal!

Thanks for your awesome work across the Node community ❤️

### PR DESCRIPTION
**Pull request by Weakky**
Closes #29

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -10,10 +10,9 @@ function deserializeNonError(t, value) {
 
 test('main', t => {
 	const serialized = serializeError(new Error('foo'));
-	const keys = Object.keys(serialized);
-	t.true(keys.includes('name'));
-	t.true(keys.includes('stack'));
-	t.true(keys.includes('message'));
+	t.truthy(serialized.name);
+	t.truthy(serialized.stack);
+	t.truthy(serialized.message);
```
Comment (Weakky) [inline test.js:15]: Not sure if this is good enough

File: index.js [hunk 2]
```diff
@@ -3,7 +3,10 @@
 class NonError extends Error {
 	constructor(message) {
 		super(NonError._prepareSuperMessage(message));
-		this.name = 'NonError';
+		Object.defineProperty(this, 'name', {
+			value: 'NonError',
+			enumerable: false
```
Comment (sindresorhus) [inline index.js:8]: This is the default.

File: test.js [hunk 3]
```diff
@@ -172,3 +171,17 @@ test('should deserialize plain object', t => {
...
+		message: 'error message',
+		stack: 'at <anonymous>:1:13',
+		name: 'name',
+		code: 'code'
+	};
+
+	const deserialized = deserializeError(object);
+	const enumerableProps = Object.keys(deserialized);
+
+	t.true(enumerableProps.length === 0);
```
Comment (sindresorhus) [inline test.js:186]: Use `t.is`

File: test.js [hunk 4]
```diff
@@ -172,3 +171,17 @@ test('should deserialize plain object', t => {
 	t.is(deserialized.name, 'name');
 	t.is(deserialized.code, 'code');
 });
+
+test('common properties should not be enumerable', t => {
+	const object = {
+		message: 'error message',
+		stack: 'at <anonymous>:1:13',
+		name: 'name',
+		code: 'code'
```
Comment (sindresorhus) [inline test.js:180]: `code` is enumerable, at least in Node.js:

```js
> try {
... 	fs.readFileSync('sadasd')
... } catch (err) {
... 	console.log(Object.keys(err));
... }
[ 'errno', 'syscall', 'code', 'path' ]
```

**Comment by sindresorhus** [comment]
Can you explicitly mention in the readme which properties are enumerable and which are not?

File: test.js (hunk 1 above)
Comment (sindresorhus) [reply test.js:15]: I would do `t.true('message' in serialized);`

File: index.js (hunk 2 above)
Comment (sindresorhus) [reply index.js:8]: However, it should be:

```
{value: "Error", writable: true, enumerable: false, configurable: true}
configurable: true
enumerable: false
value: "Error"
writable: true
__proto__: Object
```

File: test.js (hunk 4 above)
Comment (Weakky) [reply test.js]: Ah, good catch!

**Comment by Weakky** [comment]
Review done, I made `code` enumerable and adjusted the test so that it also makes sure that enumerable props are kept enumerable.

Readme + TS docs are also updated

**Pull request merged**